File: codes/C/Chapter 5/Example_5.2.c
#include "Example_5.2.h"

double Polynomial_Evaluation(double* a , int n , int x)
{
    double result = 0;
    for(int i = n - 1; i >= 0 ; --i)
        result = result * x + *(a + i);
    return result;
}

File: codes/python/Chapter 5/Example 5.2.py
# !/usr/bin/python
# -*- coding:utf-8 -*-
# Copyright 2019 Xinyi
# Author  : Xinyi Hu
# Contact : samaritanhu@gmail.com
# Time    : 2019/08/01 13:38
def Polynomial_Evaluation(a, x):
    P = a[-1]
    for i in range(1, len(a)):
        P = x*P + a[-1-i]
    return P

def run():
    a = [2, 1]
    print('P is 2 + x')
    print('The result of P(1) is: ', Polynomial_Evaluation(a, 1))

run()
